### DISCUSSION TITLE
Find all source vertices?

### DISCUSSION THREAD

**Discussion by jaufderheide** [Gremlinq/ExRam.Gremlinq#606] [Q&A] [answered]
Trying to retrieve source vertices from the CosmosDb graph. The Gremlin query is just:

g.V().not(inE())

So get all vertices with no incoming edges.

Cant figure out what Gremlinq expects inside Not(); its not used in the examples as far as I have found.

TIA if anyone can point me in the right direction! Sorry if this is a dumb question; I've been searching without much luck.

**Comment by danielcweber** [comment, marked as answer]
There's tons of usage examples in the unit tests, such as [this one](https://github.com/ExRam/ExRam.Gremlinq/blob/6b8e980bae1c592e6a28069551323439350a7d34/test/ExRam.Gremlinq.Core.Tests/QueryExecutionTest.cs#L2091).

**Reply by jaufderheide** [reply]
Thank you! Easy enough. Didn't think to look at the tests, probably a sad comment on my development experience ;)
